presto-main/src/main/java/com/facebook/presto/server/remotetask/TaskInfoFetcher.java

```diff
@@ -214,8 +214,10 @@ public void success(TaskInfo newValue)
     {
         try (SetThreadName ignored = new SetThreadName("TaskInfoFetcher-%s", taskId)) {
             lastUpdateNanos.set(System.nanoTime());
-            updateStats(currentRequestStartNanos.get());
 
+            synchronized (this) {
+                updateStats(currentRequestStartNanos.get());
```

Comment:
Oh I see, I had this wrong.  You are trying to protect `currentRequestStartNanos.get()`, so ignore my comment above. 

In this case, I think we want to minimize the scope of the `synchronized` to the  `get` call and not include all of the expensive calls for `Duration` and updating the stats.  So maybe code like this:
```java
long startNanos;
synchronized (this) {
    startNanos = currentRequestStartNanos.get();
}
updateStats(startNanos);
```

Or even better add a `private synchronized` method to get the `currentRequestStartNanos`